sys/shell/commands/sc_vfs.c

```diff
@@ -227,6 +228,113 @@ static int _read_handler(int argc, char **argv)
     return 0;
 }
 
+static int _write_handler(int argc, char **argv)
+{
+    char *w_buf;
+    char buf[16];
+    size_t nbytes = 0;
+    size_t nb_str = 0;
+    char *path = argv[1];
+    int ascii = 0;
+    int flag = O_CREAT;
+    if (argc < 2) {
+        puts("vfs write: missing file name");
+        return 1;
+    }
+
+    if (argc < 3) {
+        puts("vfs write: missing format");
+        return 1;
+    }
+    if (strcmp(argv[2], "ascii") == 0) {
+        ascii = 1;
+    }
+    else if (strcmp(argv[2], "hex") == 0) {
+        ascii = 0;
+    }
+    else {
+        printf("vfs write: unknown format: %s\n", argv[2]);
+        return 1;
+    }
+
+    if (argc < 4) {
+        puts("vfs write: missing <a|o> flag");
+        return 1;
+    }
+    if (strcmp(argv[3], "a") == 0) {
+        flag |= O_WRONLY | O_APPEND;
+    }
+    else if (strcmp(argv[3], "o") == 0) {
+        flag |= O_WRONLY;
+    }
+    else {
+        printf("vfs write: invalid flag %s\n", argv[3]);
+        return 1;
+    }
+
+    if (argc < 5) {
+        puts("vfs write: missing data");
+        return 1;
+    }
+    w_buf = argv[4];
+    nbytes = strlen(w_buf);
+    if (!ascii) {
+        /* in hex string mode, bytes may be seperated by spaces */
+        /* we need the total number of strings to go through */
+        nb_str = argc - 4;
+    }
+
+    int res;
+    res = vfs_normalize_path(path, path, strlen(path) + 1);
+    if (res < 0) {
+        _errno_string(res, (char *)buf, sizeof(buf));
+        printf("Invalid path \"%s\": %s\n", path, buf);
+        return 5;
+    }
+
+    int fd = vfs_open(path, flag, 0);
+    if (fd < 0) {
+        _errno_string(fd, (char *)buf, sizeof(buf));
+        printf("Error opening file \"%s\": %s\n", path, buf);
+        return 3;
+    }
+
+    if (ascii) {
+        res = vfs_write(fd, w_buf, nbytes);
+        if (res < 0) {
+            _errno_string(res, (char *)buf, sizeof(buf));
+            printf("Write error: %s\n", buf);
+            vfs_close(fd);
+            return 4;
+        }
+    }
+    else {
+        while (nb_str > 0) {
+            w_buf = argv[argc - nb_str];
+            nbytes = strlen(w_buf);
+            while (nbytes > 0) {
+                unsigned tmp;
+                uint8_t byte;
+                sscanf(w_buf, "%2x", &tmp);
```

Comment:
We need some sanity checks here for badly formatted inputs.
For example:
```
vfs w /dev/fram0 hex o asdf
```
results in writing two bytes: 0x0a, 0xdf